### ISSUE TITLE
memcached Could not install memcached on PHP 8.2.18

### ISSUE THREAD

**Issue by iaibai** [shivammathur/setup-php#839]
We're getting the error "Could not install memcached on PHP 8.2.18".

We're using php-version: 8.2, but we did try php-version: 8.3 and it **does** work on that.

This did seem to work yesterday (25th April), but it hasn't worked at all today.

```
==> Setup PHP
✓ PHP Installed PHP 8.2.18

==> Setup Extensions
✓ soap Installed and enabled
✗ memcached Could not install memcached on PHP 8.2.18
✓ bcmath Installed and enabled
```

Our yml is:
```
      - name: Setup PHP
        uses: shivammathur/setup-php@v2
        with:
          php-version: 8.2
          extensions: soap, memcached, bcmath
```

**Version**
- [ ] I have checked releases, and the bug exists in the latest patch version of `v1` or `v2`.
- [x] `v2`
- [ ] `v1`

**Runners**

- [x] GitHub Hosted
- [ ] Self Hosted

**Operating systems**
ubuntu-latest

**PHP versions**
PHP 8.2

**To Reproduce**
```
    steps:
      - uses: actions/checkout@v3
        with:
          ref: ${{ github.event.pull_request.head.sha }}
      - name: Setup PHP
        uses: shivammathur/setup-php@v2
        with:
          php-version: 8.2
          extensions: soap, memcached, bcmath
```

**Expected behavior**
memcached should not fail to install.

**Comment by shivammathur** [comment]
@iaibai Why os are you facing this issue on.

**Comment by iaibai** [comment]
It's ubuntu / ubuntu-latest if I understand correctly?

```
php_test_and_lint:
    if: github.event.pull_request.draft == false
    runs-on: ubuntu-latest
    container: ubuntu
```

**Comment by shivammathur** [comment]
@iaibai 

It is not working since ubuntu 24.04 was released 2 days ago and container:ubuntu now points to that. It will take some take for the extension builds to be available for that. Please either remove the container line and directly run on github hosted runners or fix the container version to 22.04 `container: ubuntu:22.04`.

**Comment by iaibai** [comment]
Thank you for looking into that. 👍
